src/main.c

```diff
@@ -366,223 +363,176 @@ int main(int argc, char* argv[]) {
-          if (!jv_is_valid(v)) {
-            fprintf(stderr, "%s: invalid JSON text passed to --argjson\n", progname);
+        } else if (isoption(&text, 0, "tab", is_short)) {
+          dumpopts &= ~JV_PRINT_INDENT_FLAGS(7);
+          dumpopts |= JV_PRINT_TAB | JV_PRINT_PRETTY;
+        } else if (isoption(&text, 0, "indent", is_short)) {
+          if (i >= argc - 1) {
+            fprintf(stderr, "%s: --indent takes one parameter\n", progname);
             die();
           }
-          program_arguments = jv_object_set(program_arguments, jv_string(argv[i+1]), v);
-        }
-        i += 2; // skip the next two arguments
-        continue;
-      }
-      if (isoption(argv[i], 0, "rawfile", &short_opts) ||
-          isoption(argv[i], 0, "slurpfile", &short_opts)) {
-        int raw = isoption(argv[i], 0, "rawfile", &short_opts);
-        const char *which;
-        if (raw)
-          which = "rawfile";
-        else
-          which = "slurpfile";
-        if (i >= argc - 2) {
-          fprintf(stderr, "%s: --%s takes two parameters (e.g. --%s varname filename)\n", progname, which, which);
-          die();
-        }
-        if (!jv_object_has(jv_copy(program_arguments), jv_string(argv[i+1]))) {
-          jv data = jv_load_file(argv[i+2], raw);
-          if (!jv_is_valid(data)) {
-            data = jv_invalid_get_msg(data);
-            fprintf(stderr, "%s: Bad JSON in --%s %s %s: %s\n", progname, which,
-                    argv[i+1], argv[i+2], jv_string_value(data));
-            jv_free(data);
-            ret = JQ_ERROR_SYSTEM;
-            goto out;
+          char* end = NULL;
+          errno = 0;
+          long indent = strtol(argv[i+1], &end, 10);
+          while (*end && isspace((unsigned char)*end))
+            end++;
+          if (errno || indent < -1 || indent > 7 || *end) {
+            fprintf(stderr, "%s: --indent takes a number between -1 and 7\n", progname);
+            die();
+          }
+          dumpopts &= ~(JV_PRINT_TAB | JV_PRINT_INDENT_FLAGS(7));
+          dumpopts |= JV_PRINT_INDENT_FLAGS(indent);
+          i++;
+        } else if (isoption(&text, 0, "seq", is_short)) {
+          options |= SEQ;
+        } else if (isoption(&text, 0, "stream", is_short)) {
+          parser_flags |= JV_PARSE_STREAMING;
+        } else if (isoption(&text, 0, "stream-errors", is_short)) {
+          parser_flags |= JV_PARSE_STREAMING | JV_PARSE_STREAM_ERRORS;
+        } else if (isoption(&text, 'e', "exit-status", is_short)) {
+          options |= EXIT_STATUS;
+        } else if (isoption(&text, 0, "args", is_short)) {
+          further_args_are_strings = 1;
+          further_args_are_json = 0;
+        } else if (isoption(&text, 0, "jsonargs", is_short)) {
+          further_args_are_strings = 0;
+          further_args_are_json = 1;
+        } else if (isoption(&text, 0, "arg", is_short)) {
+          // FIXME: For --arg and --argjson we should check that the varname is acceptable
```

Comment:
#3095

Reply:
I was thinking of removing it for the same reason, since rhe comment predates $ARGS. However, I looked into it and saw that syntactically invalid names are still added to the bytecode. Should there be validation for that or does it not matter?

Reply:
Turns out I was wrong in assuming that named args are in the bytecode of a program, so I see no need to add validation:
```
$ jq -n 42 --arg '!bad' 12345 --debug-dump-disasm
0000 TOP
0001 LOADK 42
0003 RET

42
```
I've pushed a commit removing the comment and updating the manual, and I reverted the change to the text of the comment in my refactor.

We should be good to merge now!